### DISCUSSION TITLE
MIDI input device not recognized

### DISCUSSION THREAD

**Discussion by BrianInTexas** [GrandOrgue/grandorgue#2066] [Help]
I have a working VPO built on Windows 10 LTSC and Hauptwerk.
I am trying to install and configure GO.
The MIDI input device is a Behringer FCA610.  It's driver is installed and works with Hauptwerk.
When GO starts, it sees no MIDI input device.
Any suggestions?

**Comment by midimusic** [comment]
If your midi input device is already in use by Hauptwerk then you will not be able to use it for GO. A midi input device can only be opened once.

csw900

**Comment by oleg68** [comment]
@BrianInTexas could you post the screenshot of File->Settings-Midi here?

**Reply by BrianInTexas** [reply]
Thanks for replying.
File-> in which application?
Not found in GrandOrgue. or anywhere else.
.

**Reply by oleg68** [reply]
In GrandOrgue

**Reply by BrianInTexas** [reply]
In my copy of GrandOrgue, MIDI settings are in Audio->MIDI/MIDI Devices, not in File->
![GO](https://github.com/user-attachments/assets/9e398011-83f6-4117-b7e7-a576c75203e1)
Also, no other applications are running at this time.
This used to work with the original Win 10 LTSC 2019 copy of the operating system, with HauptWerk, but never with GrandOrgue.
When I updated it to its current version, MIDI stopped working at all.
I also tried it with 10 LTSC 2021, and with 10 Pro, but none worked.
For some reason, my 10 Pro working computer does recognize MIDI devices, but that is not in the organ.
I tried to transplant that into the organ, but it was not compatible. (I do know how this works.)

**Reply by oleg68** [reply]
Thank you.

You are using a very old version of GrandOrgue.
Could you download and test the recent one - 3.15.3-1?

**Reply by BrianInTexas** [reply]
I did, and the result is identical.
Other testing I have done this morning indicates that it will likely work if I upgrade to Win 11.
I will do that today.
Thanks.

**Reply by oleg68** [reply]
> I did, and the result is identical.

But the screenshot would be different with some additional information.

**Reply by BrianInTexas** [reply]
I did look at it, and it still said there are no midi devices for both input and output.
It honestly looked identical to me.
I am preparing the Win 11 upgrade file now.

**Reply by midimusic** [reply]
I will be surprised if upgrading to windows 11 makes any difference. If here really is a problem I suspect it is in the FCA610 driver.
csw900

**Reply by oleg68** [reply]
> I did look at it, and it still said there are no midi devices for both input and output. It honestly looked identical to me. I am preparing the Win 11 upgrade file now.

The MIDI tab from recent GO versions has more fields containing some important information. So give me the screenshot exactly from 3.15.3-1.

**Reply by BrianInTexas** [reply]
Well.  Win 11 was an interesting journey.  I needed to do it anyway as part of deciding what to do with other computers.
Turns out, the real problem was Remote Desktop.  While you are connected by Remote Desktop, MIDI devices are disabled from doing anything, including being seen by other software (i.e. GrandOrgue) that you are running.  While using the computer in person, it works fine.  So, I reverted back to the LTSC 2021 version, and it works the same way.  I am not going to revert all the way to LTSC 2019, though.

The requested capture:
![Capture](https://github.com/user-attachments/assets/d0ee6cf9-996d-4181-9914-2857a6304651)

With LTSC the process count is 80, which is about half that of Win 11.

@midimusic was right about the FCA610 driver and Win 11.  It is uninstallable on Win 11.

**Comment by BrianInTexas** [comment]
So the final answers are:
1. Any Remote Desktop connection to a Windows 10 or Windows 11 24H2 computer with MIDI devices results in the MIDI devices becoming unusable until the computer is rebooted.
 
2. Microsoft is currently working to overhaul the MIDI handling with a new version, apparently backwards compatible (1/0, and with a new (2.0) mode that is intended to eventually replace 1.0 and provide much higher performance and less weirdness.

https://devblogs.microsoft.com/windows-music-dev/windows-midi-services-oct-2024-update/

3. Windows 11 Pro does not look (to me) very suitable for use in a VPO to me.  I am watching the progress of Windows 11 IoT LTSC.

4. Many MIDI devices have poor Windows support, mostly because Windows is not currently very friendly to MIDI.
